### DISCUSSION TITLE
HmIP-HAP/HmIPW-DRAP Support Patch in Kombination mit HomeAssistant Add On

### DISCUSSION THREAD

**Discussion by dn-gt** [OpenCCU/OpenCCU#2468] [🇩🇪 Fragen&Antworten]
Hallo,

zunächst vielen Dank für die Bereitstellung des [Patches ](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch) für die Möglichkeit, einen HmIPW-DRAP über das Addon in Home Assistant anzubinden.

Meine Konfiguration:
Hardware: ODroid N2+
Router: Fritz Box Fiber 5530 (192.168.0.1) - Statischer-IP-Bereich: 192.168.0.2-20
HomeAssistant OS 2023.9.2 - HomeAssistant 2023.10.1, erreichbar unter der statischen Adresse 196.168.0.2
Raspberrymatic als AddOn installiert und eingerichtet

Ich habe das Bash-Script lokal im HomeAssistant abgelegt, da ich auf der Baustelle noch kein Internet habe.

> cat patch.sh | bash

Wenn ich das Script dann ausführe, findet er die Netzwerkparameter leider nicht automatisch (ip: can't find device) und ich gebe diese daher manuell an. Soweit ich die Anleitung verstanden habe, sollte das Script eine separate Raspberrymatic-Instanz bereistellen unter der IP-Adresse, die ich als RaspberryMatic Add-on IP angebe (bei mir 192.168.0.3). Allerdings kann ich nach Durchlauf des Scripts die IP weder über den Webserver erreichen noch anpingen. Was mache ich falsch?

> cat patch.sh | bash
> ip: can't find device
> HomeAssistant Main Gateway (e.g. 192.168.178.1): 192.168.0.2
> RaspberryMatic Add-on Hostname (e.g. 5422eb72-raspberrymatic): de838cd8-raspberrymatic
> RaspberryMatic Add-on IP (e.g. 192.168.178.4): 192.168.0.3
> Creating docker macvlan network
> every ip-range or gateway must have a corresponding subnet
> Connecting add-on to macvlan network
> Error response from daemon: network ccu not found
> Stopping add-on (addon_de838cd8_raspberrymatic)
> addon_de838cd8_raspberrymatic
> Starting add-on (addon_de838cd8_raspberrymatic)
> addon_de838cd8_raspberrymatic

Die HomeAssistant Installation ist auf der statischen IP 192.168.0.2 - das ist doch vermutlich diejenige, die unter "HomeAssistant Main Gateway", angegeben werden muss? Auch mit der Router-IP habe ich es probiert, jedoch dasselbe Verhalten.

Kann es an der Fehlermeldung Network CCU not found liegen?

**Comment by Thorsten1982** [comment]
Hallo also bei mir hat der Patch funktioniert, ich wüsste nur leider nicht woran es bei dir liegen könnte. Ich beschriebe kurz mein vorgehen.

Habe HomeAssistent auf einen Pi installiert und Raspberrymatic als Addon installiert. Ip-Adresse über DCHP.

Dann wie beschrieben über SSH & Web Terminal den Befehl ausgeführt vorher natürlich den gesicherten Modus deaktiviert. 

Dann gebe ich den Hostnamen vom Raspberrymatic Addon ein und die neue IP-Adresse also einfach am Ende z.B. 100´. Danach konnte ich den DRAP anlernen. Nur meine Wandtaster und Raumthermostate werden einfach nicht gefunden weiß nicht wieso, der Falmot C12 wird gefunden. Weiß noch nicht woran es liegt.

**Reply by bf8392** [reply]
Hallo, kennt jemand eine Möglichkeit diesen Patch https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch in Home Assistant zu automatisieren bzw. eine Regel zu schreiben, so dass er quasi persistent wird?

**Reply by jens-maus** [reply]
Nein, es gibt keine möglichkeit das zu automatisieren weil der Patch ja auf der untersten Ebene von HomeAssistant - d.h. nicht innerhalb eines Addons – umgesetzt werden muss. Und von einem Addon aus hat man aus Sicherheitsgründen keinerlei möglichkeit Änderungen am grundlegenden Betriebssystem vorzunehmen. Da hilft leider weiterhin nur abwarten und hoffen das die HomeAssistant Entwickler die `macvlan` Unterstützung für HA Addons irgendwann direkt umsetzen damit dieser manuelle Patch nicht mehr notwendig wird.

**Reply by horstvogel** [reply]
> Hallo, kennt jemand eine Möglichkeit diesen Patch https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch in Home Assistant zu automatisieren bzw. eine Regel zu schreiben, so dass er quasi persistent wird?

Moin Bf8392, funktioniert das bei Dir noch mit dem Patch? 
ich bekomme seit dem Juni Update folgende Fehlermeldung. 
https://github.com/jens-maus/RaspberryMatic/discussions/2765

**Reply by horstvogel** [reply]
Hallo derzeit funktioniert das mit dem Patch bei mir nicht mehr (2 unterschiedliche Installationen, beide HA OS) 
zuerst scheint es für einen kurzen Moment zu funktionieren, dann blinkt der HMIP-HAP türkis. 

das Adon ist lokal dann außerhalb von HA erreichbar über die eingegebene IP, Also ist das macvlan dann grundsätzlich aufgespannt. So würde ich das sehen. 

Geht das noch bei jemanden? 
Vielen dank! 

https://github.com/OpenCCU/OpenCCU/wiki/Installation-HomeAssistant#hmiphaphmipwdrap-support-patch

**Reply by horstvogel** [reply]
https://github.com/OpenCCU/OpenCCU/issues/3864
